src/bin/migrate.rs

```diff
@@ -764,6 +764,61 @@ fn migrations() -> Vec<Migration> {
             try!(tx.execute("DROP INDEX users_gh_id", &[]));
             Ok(())
         }),
+        Migration::add_table(20161115110541, "categories", " \
+            id               SERIAL PRIMARY KEY, \
+            category         VARCHAR NOT NULL UNIQUE, \
+            crates_cnt       INTEGER NOT NULL DEFAULT 0, \
+            created_at       TIMESTAMP NOT NULL DEFAULT current_timestamp"),
+        Migration::add_table(20161115111828, "crates_categories", " \
+            crate_id         INTEGER NOT NULL, \
+            category_id      INTEGER NOT NULL"),
+        foreign_key(20161115111836, "crates_categories", "crate_id", "crates (id)"),
+        Migration::run(20161115111846, " \
+            ALTER TABLE crates_categories \
+            ADD CONSTRAINT fk_crates_categories_category_id \
+            FOREIGN KEY (category_id) REFERENCES categories (id) \
+            ON DELETE CASCADE", " \
```

Comment:
This just means if we delete a category it'll auto-delete everything from `crates_categories`, right?